includes/shortcodes/class-wc-shortcode-products.php

```diff
@@ -593,7 +593,7 @@ protected function get_query_results() {
 		// Remove ordering query arguments which may have been added by get_catalog_ordering_args.
 		WC()->query->remove_ordering_args();
 
-		return $results;
+		return apply_filters( 'woocommerce_shortcode_products_query_results', $results );
```

Comment:
Please update it to:

```
/**
 * Filter shortcode products query results.
 *
 * @since 4.0.0
 * @param stdClass $results Query results.
 * @param WC_Shortcode_Products $this WC_Shortcode_Products instance.
 */
return apply_filters( 'woocommerce_shortcode_products_query_results', $results, $this );
```